chia/simulator/block_tools.py

```diff
@@ -903,18 +1006,22 @@ def get_consecutive_blocks(
                         else:
                             pool_target = PoolTarget(self.pool_ph, uint32(0))
 
-                        new_gen = self.setup_new_gen(
-                            tx_block_heights,
-                            curr,
-                            wallet,
-                            rng,
-                            available_coins,
-                            prev_tx_height=prev_tx_height,
-                            dummy_block_references=dummy_block_references,
-                            transaction_data=transaction_data,
-                            include_transactions=include_transactions,
-                            block_refs=block_refs,
-                        )
+                        if new_gen_cache is None:
```

Comment:
maybe this cache should be moved into `setup_new_gen()`, since it's duplicated now